### DISCUSSION TITLE
Delay tolerant messaging blackhole

### DISCUSSION THREAD

**Discussion by zbalkan** [qaul/qaul.net#599] [Q&A] [answered]
I was watching the presentation camp 2023 and wondered the issue on delay tolerant messaging. It's basically like SMTP, you need a relay server but it is distributed on a voluntary basis.

My question is, what prevents a malicious user to create at least one account, being volunteer for relaying a message but instead, the message is dropped at this point. Therefore, the attacker can create multiple accounts to be volunteer for message relaying, creating a blackhole for these messages.

Are there any protection measures in the protocol to detect these types of malicious activities?

**Comment by MathJud** [comment, marked as answer]
## How does qaul DTN Work?

We have currently two different algorithms to decide which user will be chosen to be the storage user by the sending party.

A) **Manual Storage User Selection**: Users can manually define storage users, where they want the messages be stored for DTN.
B) **Social Proximity**: Algorithm (B) chooses storage users based on social proximity. If the sending and the receiving user are in a messaging group with another available user, this user is chosen as a storage user. 

Each message is only sent to one storage user. If algorithm (A) provides a reachable user, then this one is used, otherwise algorythm (B) is checked. If both don't find a reachable user, the sending mechanism rechecks every five minutes until the message is delivered.

## Our Plans

We want to build a multi-hop, multi-route system, which shall be based on the route-definitions of the receiving user.


## How Likely is a Blackhole?

As the DTN storage user is not chosen randomly, but either on social proximity or manually, a blackhole as a consequence of many malicous accounts is therefore not very likely for the DTN system.


## Further Malicious User Protection

This is an interesting, important and also large topic.

Currently, we have the possibility to "block" users manually. This is something each user has to do separately. We have ongoing discussions onto which depth such a blockage should go and there are plans to make it more granular.

There is no distributed blocking/trust system in place currently. But certainly an important topic for further development, for spam protection and also routing reasons.

**Reply by zbalkan** [reply]
Thank you for the detailed response. I will be following the project closely. Great job!
